internal/controller/util/vm_util.go

```diff
@@ -57,35 +66,263 @@ func ListVMsByVMNamespace(
 	log logr.Logger,
 	vmNamespaceList []string,
 	vmList []string,
-) ([]string, error) {
-	var foundVMList []string
+) []virtv1.VirtualMachine {
```

Comment:
(orthogonal to current PR) So vmlist is a list of VM names across namespaces? What if someone wanted to protect VM1/NS1 and VM2/NS2, but not VM1/NS2? With the current scheme VM1 would be protected in both namespaces afaict.